jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -0,0 +1,271 @@
+
+
+    def _on_viewer_data_changed(self, msg=None):
+        """
+        Callback method for when data is added or removed from a viewer, or
+        when a subset is created, deleted, or updated. This method receieves
+        a glue message containing viewer information in the case of the former
+        set of events, and updates the available data list displayed to the
+        user.
+
+        Notes
+        -----
+        We do not attempt to parse any data at this point, at it can cause
+        visible lag in the application.
+
+        Parameters
+        ----------
+        msg : `glue.core.Message`
+            The glue message passed to this callback method.
+        """
+        self._viewer_id = self.app._viewer_item_by_reference(
+            'spectrum-viewer').get('id')
+
+        # Subsets are global and are not linked to specific viewer instances,
+        # so it's not required that we match any specific ids for that case.
+        # However, if the msg is not none, check to make sure that it's the
+        # viewer we care about.
+        if msg is not None and msg.viewer_id != self._viewer_id:
+            return
+
+        viewer = self.app.get_viewer('spectrum-viewer')
+
+        self._viewer_data = self.app.get_data_from_viewer('spectrum-viewer')
+
+        self.dc_items = [layer_state.layer.label
+                         for layer_state in viewer.state.layers]
+
+        self.update_ui()
+
+    def update_ui(self):
+        """
+        Set up UI to have all values of currently visible spectra.
+        """
+        if len(self.dc_items) < 1:
+            self.selected_data = ""
+            self.current_flux_unit = ""
+            self.current_spectral_axis_unit = ""
+            return
+
+        self.selected_data = self.app.get_viewer("spectrum-viewer").state.reference_data.label
+
+        self.spectrum = self._viewer_data[self.selected_data]
+
+        # Set UI label to show current flux and spectral axis units.
+        self.current_flux_unit = self.spectrum.flux.unit.to_string()
+        self.current_spectral_axis_unit = self.spectrum.spectral_axis.unit.to_string()
+
+        # Populate drop down with all valid options for unit conversion.
+        self.spectral_axis_unit_equivalencies = self.create_spectral_equivalencies_list()
+        # self.flux_unit_equivalencies = self.create_flux_equivalencies_list()
+        self.flux_unit_equivalencies = self._locally_defined_flux_units()
+
+    def vue_unit_conversion(self, *args, **kwargs):
+        """
+        Runs when the `apply` button is hit. Tries to change units if `new` units are set
+        and are valid.
+        """
+
+        print(self.new_spectral_axis_unit)
+        print(self.new_flux_unit)
+        set_spectral_axis_unit = self.spectrum.spectral_axis
+        set_flux_unit = self.spectrum.flux
+
+        # Try to set new units if set and are valid.
+        if self.new_spectral_axis_unit != "" and\
```

Comment:
I can't currently change just the flux or spectral axis unit; I get a TypeError: None is not a valid Unit. Seems like this might also want to check for None?

```suggestion
        if self.new_spectral_axis_unit is not None and self.new_spectral_axis_unit != "" and\
```